### PR TITLE
[SPARK-45681][UI] Clone a js version of UIUtils.errorMessageCell for consistent error rendering on UI 

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

The task table on the stage page has its own version of error rendering using js, which does not catch up with recent updates on the java side.

In this PR, we cloned a js version of UIUtils.errorMessageCell to make them behave the same

### Why are the changes needed?

UI improvement

### Does this PR introduce _any_ user-facing change?

yes, errors in the task table on the stage page will be abbreviated to error class first

### How was this patch tested?

locally verified

#### before

image

#### after

image

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

File: core/src/main/resources/org/apache/spark/ui/static/utils.js [hunk 1]
```diff
@@ -231,4 +231,42 @@ function createRESTEndPointForMiscellaneousProcess(appId) {
...
+
+const ERROR_CLASS_REGEX = /\[([A-Z][A-Z_.]+[A-Z])\]/;
+
+/* This function works exactly the same as UIUtils.errorSummary, it shall be
+ * remained the same whichever changed */
+function errorSummary(errorMessage) {
+  let isMultiline = true;
+  const maybeErrorClass = errorMessage.match(ERROR_CLASS_REGEX);
+  let errorClassOrBrief;
+  if (maybeErrorClass) {
```
Comment (sarutak) [inline core/src/main/resources/org/apache/spark/ui/static/utils.js:254]: Is it better to check the length of `maybeErrorClass` is enough?

**Review by sarutak — COMMENTED** [review]
@yaooqinn 
Sorry for the late reply. This change almost LGTM.
Could you show the screenshots in the description to help easily understand the difference between before and after this change?

**Comment by yaooqinn** [comment]
Thank you @sarutak. I updated the PR desc with UI changes before and after this PR and addressed your comments. Please take a look again when you have some time.

**Review by sarutak — APPROVED** [review]
LGTM. Pending GA.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by yaooqinn** [comment]
Thank you @sarutak and @dongjoon-hyun.

Test passed. Merged to master

**Pull request closed without merging**